backend/src/v2/compiler/argocompiler/importer.go

```diff
@@ -76,10 +77,8 @@ func (c *workflowCompiler) addImporterTemplate() string {
 		fmt.Sprintf("$(%s)", component.EnvPodName),
 		"--pod_uid",
 		fmt.Sprintf("$(%s)", component.EnvPodUID),
-		"--mlmd_server_address",
-		fmt.Sprintf("$(%s)", component.EnvMetadataHost),
-		"--mlmd_server_port",
-		fmt.Sprintf("$(%s)", component.EnvMetadataPort),
+		"--mlmd_server_address", common.GetMetadataGrpcServiceServiceHost(),
+		"--mlmd_server_port", common.GetMetadataGrpcServiceServicePort(),
```

Comment:
It's worth noting that in this pr we seem to have ignored importer launcher entirely
I can't recall if it was an intentional decision on our part or not, do you remember? 

importer launcher will also create an mlmd client, so it seems like it should be part of this Jira, we can do it as a follow up item/pr, I don't want to necessarily block this pr on this. 

I'm not entirely certain if importer even works properly.

Since it should be mostly just leveraging the same logic you've already added, I can't imagine it's going to be that hard to add.

Reply:
Jira created: https://issues.redhat.com/browse/RHOAIENG-13327